### PR TITLE
refactor(presenter): make input fields and checkbox reusable components

### LINKED ISSUE #91: 🔧 [Refactor] Use and reuse components for Input Fields
## 🔧 Refactor

We have duplicate code for name and email input fields.

`presenter/src/components/inputs/ContactForm.vue`

`presenter/src/components/inputs/NewsletterForm.vue`

 * name input field
 * email input field
 * data privacy checkbox input field

### LINKED ISSUE #155: 🔧 [Refactor] Improve components and component reusability
## 🔧 Refactor

Improve components and their reusability.

We have a specific issue for Inputfields: #91 .

Please only make practical improvements - we do not need theoretical improvements where a component might be reusable.

### PR DESCRIPTION
**Pull request by mahula**
### Issues
- fixes #91 
- fixes #155

### Todo

- [X] refactor input fields to reusable components
- [x] refactor data privacy checkbox to reusable component

### CODE REVIEW

**Review by Mogge — CHANGES_REQUESTED** [review]
```
 [vue-tsc] Property 'value' does not exist on type '{ $: ComponentInternalInstance; $data: {}; $props: Partial<{}> & Omit<{ "onModelValue:update"?: ((value: boolean) => any) | undefined; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<...> & { ...; }, never>; ... 10 more ...; $watch<T extends string | ((...args: any) => any)>(source: T, cb: T...'. 
```

**Review by Mogge — CHANGES_REQUESTED** [review]
* take snapshots of the new components
 * test that the update event is fired

**Comment by mahula** [comment]
> * take snapshots of the new components
> * test that the update event is fired

Done

**Review by Mogge — APPROVED** [review]
Great work!

File: presenter/src/components/controls/DataPrivacyCheckbox.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  beforeEach(() => {
+    wrapper = Wrapper()
+  })
+
+  it('renders', () => {
+    expect(wrapper.element).toMatchSnapshot()
+  })
+
+  it('emits the modelValue:update event on checkbox change', async () => {
+    await wrapper.find('input[type="checkbox"]').trigger('change')
```
Comment (Mogge) [inline presenter/src/components/controls/DataPrivacyCheckbox.test.ts:21]:
```suggestion
    await wrapper.find('input[type="checkbox"]').setValue(true)
```

**Review by Mogge — CHANGES_REQUESTED** [review]

File: presenter/src/components/controls/DataPrivacyCheckbox.test.ts (hunk 1 above)
Comment (Mogge) [reply presenter/src/components/controls/DataPrivacyCheckbox.test.ts:21]: https://test-utils.vuejs.org/api/#setValue

**Review by Mogge — CHANGES_REQUESTED** [review]
Please fix the warnings in the unit tests

**Comment by mahula** [comment]
> Please fix the warnings in the unit tests

Done.

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]
Works as expected.

- Why do you put `DataPrivacyCheckbox` in `controls` rather than `inputs`?
- All Stories are missing

**Comment by mahula** [comment]
> * Why do you put `DataPrivacyCheckbox` in `controls` rather than `inputs`?

Because I see it more on the control side (checkboxes, radio buttons etc.) than as an input element.

**Comment by mahula** [comment]
> Please fix the warnings in the unit tests

Done

**Review by Elweyn — COMMENTED** [review]
```
  Snapshots  2 obsolete
             ↳ src/components/sections/PlanningSection.test.ts
               · TimelineSection > renders 1
             ↳ src/components/menu/HeaderMenu.test.ts
               · TopMenu > renders three columns 1
```

**Comment by mahula** [comment]
> ```
>   Snapshots  2 obsolete
>              ↳ src/components/sections/PlanningSection.test.ts
>                · TimelineSection > renders 1
>              ↳ src/components/menu/HeaderMenu.test.ts
>                · TopMenu > renders three columns 1
> ```

Fixed

**Review by Elweyn — DISMISSED** [review]
Seems good for me!

File: presenter/src/components/forms/NewsletterForm.vue [hunk 2]
```diff
@@ -68,18 +46,7 @@
...
-            color="rgba(174, 179, 189, 0.50)"
-            value="false"
-            hide-details="auto"
-            :center-affix="false"
-            density="compact"
-            required
-            :rules="[() => !!dataprivacy || $t('validation.fieldRequired')]"
-          >
-          </v-checkbox>
+          <DataPrivacyCheckbox></DataPrivacyCheckbox>
```
[the remark this answers is not in the document]
Comment (mahula) [reply presenter/src/components/forms/NewsletterForm.vue]: Fixed

File: presenter/src/components/forms/ContactForm.vue [hunk 3]
```diff
@@ -92,18 +70,7 @@
...
-            color="rgba(174, 179, 189, 0.50)"
-            value="false"
-            hide-details="auto"
-            :center-affix="false"
-            density="compact"
-            required
-            :rules="[() => !!dataprivacy || $t('validation.fieldRequired')]"
-          >
-          </v-checkbox>
+          <DataPrivacyCheckbox></DataPrivacyCheckbox>
```
[the remark this answers is not in the document]
Comment (mahula) [reply presenter/src/components/forms/ContactForm.vue]: Fixed

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**